### ISSUE TITLE
libsForQt5*: bad interaction with nativeBuildInputs

### ISSUE THREAD

**Issue by vcunat** [NixOS/nixpkgs#98542]
_I haven't looked into details, at least for now._

I noticed that after #97524 `kdiff3` started failing to build:
```
kparts-5.73.0/lib/libKF5Parts.so.5.73.0: undefined reference to `qt_version_tag@Qt_5.14'
```
which was weird.  Modifying kdiff3 expression not to use `nativeBuildInputs` works around the problem, so apparently there's some bad interaction between `libsForQt5*.callPackage` and native mechanism.  It's well possible that the issue is more general than just Qt.

/cc @ttuegel

**Comment by FRidh** [comment]
is it the following ? https://github.com/NixOS/nixpkgs/issues/49526

**Comment by doronbehar** [comment]
This was fixed in #99956 . kdiff works fine for me now.

**Comment by doronbehar** [comment]
To be more specific: https://github.com/NixOS/nixpkgs/commit/6b6f6808318d4a445870048d1175fcb55b1b69aa .
